docs/refactor ideas.md

```diff
@@ -0,0 +1,2 @@
+- [ ] Convert dimensions to classes or introduce new properties to their types to consolidate their behavior and characteristics, instead of having a lot of separated functions w/o real connection
```

Comment:
I would put optimization and refactor in the same file

Reply:
I updated the file title to contain "performance optimizations", I think they should still be separate from technical refactors?